src/components/top-bar-equipment-seach-dialog/use-top-bar-search-matching-equipments.tsx

```diff
@@ -8,30 +8,37 @@ import { UUID } from 'crypto';
 import { useSearchMatchingEquipments } from './use-search-matching-equipments';
 import { useMemo } from 'react';
 import { getLocalStorageSearchEquipmentHistory } from 'redux/local-storage/search-equipment-history';
+import { EquipmentType } from '@gridsuite/commons-ui';
 
 interface UseTopBarSearchMatchingEquipmentProps {
     studyUuid: UUID;
     nodeUuid: UUID;
+    equipmentType?: EquipmentType;
 }
 
 export const useTopBarSearchMatchingEquipment = (
     props: UseTopBarSearchMatchingEquipmentProps
 ) => {
-    const { nodeUuid, studyUuid } = props;
+    const { nodeUuid, studyUuid, equipmentType } = props;
     const { equipmentsFound, searchTerm, ...otherStates } =
         useSearchMatchingEquipments({
             studyUuid: studyUuid,
             nodeUuid: nodeUuid,
+            equipmentType: equipmentType ?? undefined,
         });
 
     // when searching from topbar, if user input is empty, display last clicked element from localstorage
     const equipmentsToReturn = useMemo(() => {
-        if (searchTerm.length > 0) {
+        if (searchTerm.length > 0 || equipmentType != null) {
```

Comment:
maybe use undefined instead of null ?
i can see in top-bar-equipment-seach-dialog/top-bar-equipment-search-dialog.tsx undefined is passed to this hook

Reply:
it works the same way, undefined == null in JS
But I agree it's cleaner to only check "!== undefined" since it can't be null according to TS